skywalking/plugins/sw_pyramid.py

```diff
@@ -0,0 +1,56 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from skywalking import Layer, Component
+from skywalking.trace import tags
+from skywalking.trace.carrier import Carrier
+from skywalking.trace.context import get_context
+from skywalking.trace.tags import Tag
+
+
+def install():
+    from pyramid.router import Router
+
+    def _sw_invoke_request(self, request, *args, **kwargs):
+        context = get_context()
+        carrier = Carrier()
+
+        for item in carrier:
+            val = request.headers.get(item.key)
+
+            if val is not None:
+                item.val = val
+
+        with context.new_entry_span(op=request.path, carrier=carrier) as span:
+            span.layer = Layer.Http
+            span.component = Component.Pyramid
+            span.peer = request.remote_host or request.remote_addr
+
+            span.tag(Tag(key=tags.HttpMethod, val=request.method))  # pyre-ignore
+            span.tag(Tag(key=tags.HttpUrl, val=str(request.url)))  # pyre-ignore
```

Comment:
:facepalm::facepalm::facepalm:

Reply:
@TomMD can you please help to configure ignoring this in somewhere? Or how can I do it by myself?

Reply:
Its strange how these lines got flagged as an error in aiohttp but not here under almost identical circumstances.

Reply:
Here is @TomMD's suggestion from that pull:

> I think this is the second "missing argument" error on Tag.init. If you'd like I can make a PR (or you are free to) that adds igore = [ "Missing argument" ] to the config.toml. Otherwise I'm likely to do something in the backend that removes this seemingly common seemingly false positive.

Reply:
@kezhenxu94 Yes I can absolutely do that.  I'm a bit buried so if you could give me till the end of the week I'd appreciate it.